fuzzers/inprocess/libfuzzer_template_lib/harness.c

```diff
@@ -0,0 +1,123 @@
+
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/stat.h>
+#include <dirent.h>
+
+#include "template.h"
+
+#define MAX_CORPUS_SIZE 10 * 1024 * 1024 //Max 10Mo
+
+int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  int ret = template_run_some_data((char*)data, size);
+  return ret;
+
+}
+
+static int filter(const struct dirent *name){return 1;}
+static int test_one_corpus(char * corpus) {
+
+  int ret;
+  struct stat stat_;
+
+  int fd = open(corpus, O_RDONLY);
+  if (fd == -1) {
+    printf("Error opening corpus\n");
+    ret = -1;
+    goto out;
+  }
+  
+  stat(corpus, &stat_);
+
+  int corpus_size = stat_.st_size;
+  printf("Corpus: %s | Size : %d\n", corpus, corpus_size);
+
+  if (corpus_size > MAX_CORPUS_SIZE) {
+    printf("corpus size too big\n");
+    ret = -1;
+    goto out;
+  }
+
+  char * corpus_data = calloc(1, corpus_size);
+  int bsize = read(fd, corpus_data, corpus_size);
+
+  if (bsize != corpus_size) {
+    printf("corpus size doesn't match readed size (%d != %d)!\n", bsize, corpus_size);
+    ret = -1;
+    goto out;
+  }
+
+  ret = LLVMFuzzerTestOneInput((const uint8_t*)corpus_data, (size_t)corpus_size);
+  printf("ret=%d\n", ret);
+
+  out:
+
+    if (corpus_data) {
+      free(corpus_data);
+    }
+
+    if (fd > 0) {
+      close(fd);
+    }
+    return ret;
+
+}
+
+// simply RECOMPILE WITH -DTEST_CORPUS=1 OR -DTEST_ALL_CORPUS=1 to test corpus_evolution or crashes directory. 
+#ifdef TEST_CORPUS
```

Comment:
This doesn't need to be part of the harness, it's better to put an argument for rerunning a corpus into lib.rs - then you don't need to recompile the target, either.